conda_smithy/configure_feedstock.py

```diff
@@ -1872,45 +1753,109 @@ def _load_forge_config(forge_dir, exclusive_config_file, forge_yml=None):
-            documents = list(yaml.safe_load_all(fh))
-            file_config = (documents or [None])[0] or {}
 
-        # The config is just the union of the defaults, and the overriden
-        # values.
-        config = _update_dict_within_dict(file_config.items(), config)
+    with open(forge_yml, "r") as fh:
+        documents = list(yaml.safe_load_all(fh))
+        file_config = (documents or [None])[0] or {}
 
-        # check for conda-smithy 2.x matrix which we can't auto-migrate
-        # to conda_build_config
-        if file_config.get("matrix") and not os.path.exists(
-            os.path.join(
-                forge_dir, config["recipe_dir"], "conda_build_config.yaml"
-            )
-        ):
-            raise ValueError(
-                "Cannot rerender with matrix in conda-forge.yml."
-                " Please migrate matrix to conda_build_config.yaml and try again."
-                " See https://github.com/conda-forge/conda-smithy/wiki/Release-Notes-3.0.0.rc1"
-                " for more info."
-            )
+    # The config is just the union of the defaults, and the overridden
+    # values.
+    config = _update_dict_within_dict(file_config.items(), config)
 
-        if file_config.get("docker") and file_config.get("docker").get(
-            "image"
-        ):
-            raise ValueError(
-                "Setting docker image in conda-forge.yml is removed now."
-                " Use conda_build_config.yaml instead"
-            )
+    # check for conda-smithy 2.x matrix which we can't auto-migrate
+    # to conda_build_config
+    if file_config.get("matrix") and not os.path.exists(
+        os.path.join(
+            forge_dir, config["recipe_dir"], "conda_build_config.yaml"
+        )
+    ):
+        raise ValueError(
+            "Cannot rerender with matrix in conda-forge.yml."
+            " Please migrate matrix to conda_build_config.yaml and try again."
+            " See https://github.com/conda-forge/conda-smithy/wiki/Release-Notes-3.0.0.rc1"
+            " for more info."
+        )
+
+    if file_config.get("docker") and file_config.get("docker").get("image"):
+        raise ValueError(
+            "Setting docker image in conda-forge.yml is removed now."
+            " Use conda_build_config.yaml instead"
+        )
+
+    if (
+        "build_with_mambabuild" in file_config
+        and "conda_build_tool" not in file_config
+    ):
+        warnings.warn(
+            "build_with_mambabuild is deprecated, use conda_build_tool instead",
+            DeprecationWarning,
+        )
+        config["conda_build_tool"] = (
+            "mambabuild" if config["build_with_mambabuild"] else "conda-build"
+        )
+    if file_config.get("conda_build_tool_deps"):
+        raise ValueError(
+            "Cannot set 'conda_build_tool_deps' directly. "
+            "Use 'conda_build_tool' instead."
+        )
+
+    # Validate the merged configuration against a JSON schema
```

Comment:
I feel we should validate with the json schema _before_ we run the custom validations.

Reply:
Indeed, I was glancing over that so many times that I forgot, but our own values might obfuscate issues with the forge yaml present in the feedstock due to the merge of values. Thanks for pointing out

Reply:
done 😄